src/ng/compile.js

```diff
@@ -1909,7 +2001,7 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             : function denormalizeTemplate(template) {
               return template.replace(/\{\{/g, startSymbol).replace(/}}/g, endSymbol);
         },
-        NG_ATTR_BINDING = /^ngAttr[A-Z]/;
+        NG_PREFIX_BINDING = /^ng(Attr|Prop|On)([A-Z].*)$/;
```

Comment:
Nit: With this rregex it is not possible to bind to properties that don't start with a lowercase letter (e.g. `Foo`, `_bar`, `$baz`). If we decide not to support those, we should at least document the limitations.

Reply:
That means `ng-attr-*` always had that problem? :thinking:

Reply:
Yes, but is it unusual for attribute to start with an uppercase letter, `_` or `$` :smiley:
(I guess it is not very likely for properties either 🤔 - at least not for public properties you would want to bind to :wink:)

Reply:
I think I'd like to fix this. Basically just change the end of the regex to `(_?[A-Z].*)$`?

Reply:
If you want to support uppercase first letter, it should be `((?:_[a-z]|[A-Z]).*)$` (because in order to bind to `Foo`, you would need `ng-prop-_foo`, which would be normalized to `ngProp_foo`).

Reply:
What about properties starting with an `_`? 🤔

Reply:
You could bind to them with `__foo` (which is super-weird). But `_`-prefixed properties in JS usually means "private property", which in turn means you probably shouldn't be binding to it :grin:
(BTW, there is the same issue with attributes containing an underscore. Essentially, if you want to bind to attr/prop (using `ngAttr*`/`ngProp*`) that contains a `_`, you need to "escape" it wlike this: `__` (two underscores).)

Reply:
... and attributes.  Where should that be documented though?

Could also fix it though...